clang/test/SemaSYCL/address-space-parameter-conversions.cpp

```diff
@@ -12,13 +12,21 @@ void tmpl(T *t){}
 
 void usages() {
   __attribute__((opencl_global)) int *GLOB;
+  __attribute__((opencl_global_device)) int *GLOB_DEV;
+  __attribute__((opencl_global_host)) int *GLOB_HOST;
   __attribute__((opencl_private)) int *PRIV;
   __attribute__((opencl_local)) int *LOC;
   int *NoAS;
 
   bar(*GLOB);
   bar2(*GLOB);
 
+  bar(*GLOB_DEV);
```

Comment:
Following your description, it is missing address space overlap tests for `opencl_global_device`/`opencl_global_host` to `opencl_global`